test/server/http_test_server_filter_integration_test.cc

```diff
@@ -215,94 +132,60 @@ TEST_P(HttpTestServerIntegrationTest, TestEchoHeaders) {
+TEST_P(HttpTestServerIntegrationTest, TestNoStaticConfigZeroLengthRequest) {
+  initializeFilterConfiguration(kNoConfigProto);
   testWithResponseSize(0, false);
 }
 
-TEST_P(HttpTestServerIntegrationNoConfigTest, TestMaxBoundaryLengthRequest) {
+TEST_P(HttpTestServerIntegrationTest, TestNoStaticConfigMaxBoundaryLengthRequest) {
+  initializeFilterConfiguration(kNoConfigProto);
   const int max = 1024 * 1024 * 4;
   testWithResponseSize(max, false);
 }
 
-TEST_P(HttpTestServerIntegrationNoConfigTest, TestTooLarge) {
+TEST_P(HttpTestServerIntegrationTest, TestNoStaticConfigTooLarge) {
+  initializeFilterConfiguration(kNoConfigProto);
   const int max = 1024 * 1024 * 4;
   testBadResponseSize(max + 1);
 }
 
-TEST_P(HttpTestServerIntegrationNoConfigTest, TestHeaderConfig) {
-  Envoy::BufferingStreamDecoderPtr response = makeSingleRequest(
-      lookupPort("http"), "GET", "/", "", downstream_protocol_, version_, "foo.com", "",
-      [](Envoy::Http::RequestHeaderMapImpl& request_headers) {
-        const std::string header_config =
-            R"({response_headers: [ { header: { key: "foo", value: "bar2"}, append: true } ]})";
-        request_headers.addCopy(Nighthawk::Server::TestServer::HeaderNames::get().TestServerConfig,
-                                header_config);
-      });
+TEST_P(HttpTestServerIntegrationTest, TestNoStaticConfigHeaderConfig) {
+  initializeFilterConfiguration(kNoConfigProto);
+  setRequestLevelConfiguration(
+      R"({response_headers: [ { header: { key: "foo", value: "bar2"}, append: true } ]})");
+  Envoy::IntegrationStreamDecoderPtr response = getResponse(ResponseOrigin::EXTENSION);
+
   ASSERT_TRUE(response->complete());
   EXPECT_EQ("200", response->headers().Status()->value().getStringView());
   EXPECT_EQ("bar2",
             response->headers().get(Envoy::Http::LowerCaseString("foo"))->value().getStringView());
   EXPECT_EQ("", response->body());
 }
 
-TEST_P(HttpTestServerIntegrationNoConfigTest, BadTestHeaderConfig) {
-  Envoy::BufferingStreamDecoderPtr response = makeSingleRequest(
-      lookupPort("http"), "GET", "/", "", downstream_protocol_, version_, "foo.com", "",
-      [](Envoy::Http::RequestHeaderMapImpl& request_headers) {
-        request_headers.addCopy(Nighthawk::Server::TestServer::HeaderNames::get().TestServerConfig,
-                                kBadJson);
-      });
-  ASSERT_TRUE(response->complete());
-  EXPECT_EQ("500", response->headers().Status()->value().getStringView());
-  EXPECT_EQ("test-server didn't understand the request: Error merging json config: Unable to parse "
-            "JSON as proto (INVALID_ARGUMENT:Unexpected token.\nbad_json\n^): bad_json",
-            response->body());
-}
-
-class HttpTestServerDecoderFilterTest : public Test {};
-
 // Here we test config-level merging as well as its application at the response-header level.
-TEST_F(HttpTestServerDecoderFilterTest, HeaderMerge) {
+TEST(HttpTestServerDecoderFilterTest, HeaderMerge) {
```

Comment:
I might be misunderstanding how this test works. Is there any chance this could be split into a couple separate cases being tested? It feels like it's trying to test several different kind of header merge.

Reply:
You're right, this test has organically grown as different contributors added to it, and splitting this up makes sense.
As this is pre-existing, I prefer to file an issue and address this in a follow up, would you be ok with that?

Reply:
Ok.

Reply:
Filed #559